client/hosting/overview/components/plan-bandwidth.tsx

```diff
@@ -22,9 +35,19 @@ const getCurrentMonthRangeTimestamps = () => {
 };
 
 export function PlanBandwidth( { siteId }: PlanBandwidthProps ) {
+	const [ bandwidthData, setBandwidthData ] = useState< SiteMetricsAPIResponse >();
 	const selectedSiteData = useSelector( getSelectedSite );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const isAtomic = useSelector( ( state ) => isAtomicSite( state, siteId ) );
+	const canViewStat = useSelector( ( state ) => canCurrentUser( state, siteId, 'publish_posts' ) );
+	const hasSftpFeature = useSelector( ( state ) => siteHasFeature( state, siteId, FEATURE_SFTP ) );
+
+	const isEligibleForAtomic = ! isAtomic && hasSftpFeature;
+
 	const selectedSiteDomain = selectedSiteData?.domain;
 
+	const translate = useTranslate();
+
 	const { startInSeconds, endInSeconds } = getCurrentMonthRangeTimestamps();
 
 	const { data } = useSiteMetricsQuery( siteId, {
```

Comment:
Before we had check for `isAtomic`, to avoid redundant requests to the server. It doesn't make sense to overload server when it obviously returns error.
Let's disable this request for simple websites

Reply:
I don't recall having this in an `isAtomic` check (React would have hated a conditional hook call like that) but you make an excellent point about not requesting data we know isn't available.

I've encapsualted the logic for retrieving atomic site bandwidth data in a separate sub-component, so `useSiteMetricsQuery` only fires when we actually need it to, but without violating the rules of hooks. This also introduced a small change where if we don't have a valid `selectedSiteDomain` for some reason, we'll get the same "Not available" output a simple site would have, instead of a perpetual placeholder.

Reply:
I meant not conditional query, but disabling running the query in case of atomic.
https://tanstack.com/query/v4/docs/framework/react/guides/disabling-queries#:~:text=If%20you%20ever%20want%20to,'success'%20or%20isSuccess%20state.

Reply:
Oh, I see... you meant when the entire `PlanBandwidth` component was behind an `isAtomic` condition, not that the `useSiteMetricsQuery` was somehow being handled conditionally. That makes more sense, thank you for clarifying!

I'll look into utilizing `enabled: false` as a potential followup!